### PR TITLE
cmds/core/mktemp: remove testutil

### PR DESCRIPTION
**Pull request by binjip978**

### CODE REVIEW

**Comment by binjip978** [comment]
Added FlagSet idea from slack discussion.

File: cmds/core/mktemp/mktemp.go [hunk 1]
```diff
@@ -30,14 +30,21 @@ package main
...
 )
 
-type mktempflags struct {
+type cmd struct {
+	stdout io.Writer
+	flags  flags
+	args   []string
+}
+
+type flags struct {
```
Comment (10000TB) [inline cmds/core/mktemp/mktemp.go:47]: can you consider name this after sth else like `Option` ?  `flags` is almost like standard pkg `flag`

File: cmds/core/mktemp/mktemp.go [hunk 2]
```diff
@@ -47,60 +54,70 @@ type mktempflags struct {
...
+		return err
+	}
+
+	fmt.Fprintf(c.stdout, "%s\n", fileName)
+	return nil
+}
+
+func (f *flags) register(fs *flag.FlagSet) {
+	flag.BoolVarP(&f.d, "directory", "d", false, "Make a directory")
+	flag.BoolVarP(&f.u, "dry-run", "u", false, "Do everything save the actual create")
```
Comment (10000TB) [inline cmds/core/mktemp/mktemp.go:109]: Do everything except the actual creation ?

File: cmds/core/mktemp/mktemp.go [hunk 3]
```diff
@@ -47,60 +54,70 @@ type mktempflags struct {
...
+	fileName, err := c.mktemp()
+	if err != nil && !c.flags.q {
+		return err
+	}
+
+	fmt.Fprintf(c.stdout, "%s\n", fileName)
+	return nil
+}
+
+func (f *flags) register(fs *flag.FlagSet) {
```
Comment (10000TB) [inline cmds/core/mktemp/mktemp.go:107]: `fs` is not used within `register` ?

File: cmds/core/mktemp/mktemp.go (hunk 2 above)
Comment (binjip978) [reply cmds/core/mktemp/mktemp.go:109]: yep, mktemp seems like a linux thing, not a part of POSIX, but in linux man such option exists.

```
       -u, --dry-run
              do not create anything; merely print a name (unsafe)
```

File: cmds/core/mktemp/mktemp.go (hunk 1 above)
Comment (binjip978) [reply cmds/core/mktemp/mktemp.go:47]: It should be something related to flags, maybe cmdFlags or something. Options can be confusing as well, people may think that is something related to WithXXX option pattern.

Anyway it kind of experiment of moving away from globals. Not even sure if should do it at all.

File: cmds/core/mktemp/mktemp.go (hunk 3 above)
Comment (binjip978) [reply cmds/core/mktemp/mktemp.go:107]: thx, fixed

**Comment by rminnich** [comment]
can we wrap this up?

File: cmds/core/mktemp/mktemp.go (hunk 1 above)
Comment (rminnich) [reply cmds/core/mktemp/mktemp.go:47]: remember people use package names all the time for variables; you can find variables named path everywhere. I think this is descrptive.

Comment (rminnich) [reply cmds/core/mktemp/mktemp.go:47]: I'm gonna close this comment, I think it's ok.

**Review by rminnich — APPROVED** [review]

**Pull request merged**